Real Device Cleaning Process  documentation (#1557)
saucelabs/sauce-docs: docs/mobile-apps/real-device-cleaning.md

---
id: real-device-cleaning
title: Real Device Cleaning Process
sidebar_label: Real Device Cleaning Process
---
import useBaseUrl from '@docusaurus/useBaseUrl';

The cleaning process runs at the end of each test session for Sauce Labs Real Device. It resets the device back to a default state, ready for the next test. The cleaning process also removes any private data which may have been part of the prior test. If we encounter any issue during the cleaning process, or we fail to clean the device, we put those devices aside for manual inspection to prevent data and security leaks. 

> The cleaning process differs slightly between the Public Devices which are shared among all our customers, and the Private Devices, where the owner has more control over the configuration of their device. It also differs between Android and iOS, due to the different features of these operating systems.

Reviewer: ```suggestion
The cleaning process differs slightly between the Public Devices, which are shared among all our customers, and the Private Devices, where the owner has more control over the configuration of their device. It also differs between Android and iOS, due to the different features of these operating systems.
```

## Public Devices

> Every 5th cleaning session includes a device reboot. This ensures that the system remains 

Reviewer 2: ```suggestion
Every fifth cleaning session includes a device reboot. This ensures that the system remains 
```

performant, and any stray processes are stopped.

:::note

> A factory reset is not performed between test sessions, as this requires manual intervention and would slow down the time required between test runs.

Reviewer 2: This sentence is still a little hard to understand. I think it was meant to say that a factory reset would make the time between test runs take longer. 
```suggestion
A factory reset is not performed between test sessions, as this requires manual intervention and would increase the time required between tests.
```

:::

### Cleaning Process Steps

> We use a proprietary process that wipes every real device clean at the end of the testing session:

Reviewer: Do these steps always occur in this order? If so, the list could be reformatted as a numbered list.

* AppleID’s are removed between sessions.
* WiFi/network and DNS configuration settings are reset.
* System locale, language, and time zone are reset to match those of the hosting Data Center.
  * EU devices are reset to UK-English, in the CET (UTC+1) time zone.
  * US devices are reset to US-English, in the PST (UTC-8) time zone.
* Mock GPS location is reset to that of the hosting Data Center
 * US Devices: Lat 37.768, Lon -121.959, Alt 5m (San Francisco, US) and Lat 33.771 Lon -84.397, Alt 225m (Atlanta, US)
 * EU devices: Lat 52.500, Lon 13.447, Alt 40m (Berlin, DE)
* Media files (Photos, Videos, Files) on the device are removed.
* PIN code/Passcode is removed.

## Private Devices

Private Devices follow the same process as Public Devices. However, the customer can adjust the following as desired:

App allowlist:
* Apps installed on the device can be selected, by name/bundleID/package name, to be preserved between sessions. These apps and their data will not be cleaned. They can be uploaded by the customer from Sauce Storage, or installed from the App Store/Play Store.
  
Account allowlist: 
* Google accounts signed into the Play Store can be preserved between sessions.

> * Apple IDs signed in to the device and App Store can be preserved between sessions.

Reviewer 2: ```suggestion
* Apple IDs signed into the device and App Store can be preserved between sessions.
```

> ## Goals of the Cleaning Process

Reviewer: This section could be moved higher up in the page.
